R/derive_advs_params.R

```diff
@@ -514,7 +524,7 @@ derive_param_ratio <- function(dataset,
 
   ### If `get_unit_expr` provided then check units and enable units conversion ----
 
-  if (!missing(get_unit_expr) && !is.null(get_unit_expr)) {
+  if (unit_conversion == TRUE && !missing(get_unit_expr) && !is.null(get_unit_expr)) {
```

Comment:
Minor, but stylistically I'd prefer `if(unit conversion ...` rather than `if(unit_conversion == TRUE ...` throughout:

```suggestion
  if (unit_conversion && !missing(get_unit_expr) && !is.null(get_unit_expr)) {
```